### PR TITLE
Update ServerSettingsPage.cs

### LINKED ISSUE #275: [BUG] Non-MC can change the level locally for themselves
**Describe the bug**
When a non-MC attempts to change the level, upon the next restart, this level is changed locally for them. This is undesired behavior and only the MC can determine which levels are loaded.

**To Reproduce**
1. Host a lobby a MC and join with a non-MC client
2. As non-MC, go to Game Settings > Server Settings, choose a new level and select "Change"
3. As MC, type /restart
4. The level of the MC is not changed, but the level of the non-MC is not changed.

**Expected behavior**
If a player is not the master client, then they must respect the current level that is loaded and be prevented from loading other levels.

**Contributor instructions** (Do not modify)
1. Try to recreate the bug via the reported reproduction steps. If the bug can be reproduced, remove the investigation label. If the bug cannot be reproduced, please add a comment, attach "On Hold" label and inform the submitter that the bug cannot be reproduced. If there is no reply after 2 weeks, the issue can be closed.
2. Follow the default development team procedures (Create a new branch, apply your fix, create a pull request, submit new builds for testing team in dev squad)
3. Once testing team has confirmed the issue is resolved and the pull request is accepted, the issue can be closed.

### PR DESCRIPTION
**Pull request by sonicv6**

### CODE REVIEW

**Review by Lithrun — CHANGES_REQUESTED** [review]
Testing Team notices an issue with this:
```
Both players start on initial level -> MC changes level and restarts -> both players on new level (as it should be) -> non-MC changes level, MC restarts -> MC remains on its level, non-MC goes back to initial level (regardless which level it chooses)
So, in short, as soon as non-MC changes the level locally it seems to be "bound" to the level it joined in on upon each restart. Would be nice if anyone else could confirm this, but I'm pretty sure that's what's happening now.
```
Please investigate and resolve it. Will change to approved once @AoTTG-2/testing-team can confirm it works as intended

**Comment by sonicv6** [comment]
I'll investigate some more.

**Comment by SlowDeepExhalation** [comment]
I checked the latest build, the bug Eisenlocke described seems to have been fixed.  I was unable to find any other issues with level changes not being synchronized between two clients, though I was not _completely_ exhaustive; it's possible some obscure edge cases are out there with certain maps or large numbers of players or some other oddity.  No other testers have reported problems either, so here's my go-ahead to merge it.

I don't seem to have been requested to review the PR, so this comment is all I can add.

**Review by Lithrun — APPROVED** [review]

**Pull request merged**
